pkg/backend/httpstate/client/client.go

```diff
@@ -1229,6 +1229,20 @@ func (pc *Client) PatchUpdateCheckpointDelta(ctx context.Context, update UpdateI
 		updateAccessToken(token), httpCallOptions{RetryPolicy: retryAllMethods, GzipCompress: true})
 }
 
+// AppendUpdateJournalEntry appends a new entry to the journal for the given update.
+func (pc *Client) AppendUpdateJournalEntry(ctx context.Context, update UpdateIdentifier,
```

Comment:
https://github.com/pulumi/pulumi/pull/20343/files#diff-5cf44f98617c0f9bd538e0aa675425a4efd0d09fe4bed8f3c700260a98439990R1237-R1252 has the API we've so far been discussing, that's implemented in the service PR I have open as well.  Should we re-use that here instead of building something new?